pkg/cache/cache_test.go

```diff
@@ -789,7 +797,97 @@ func CacheTest(createCacheFunc func(config *rest.Config, opts cache.Options) (ca
 					Eventually(out).Should(Receive(Equal(pod)))
 					close(done)
 				})
+				type selectorsTestCase struct {
+					fieldSelectors map[string]string
+					labelSelectors map[string]string
+					expectedPods   []string
+				}
+				DescribeTable("filter informers at list watch with selectors", func(tc selectorsTestCase) {
+					By("creating the cache")
+					builder := cache.BuilderWithOptions(
+						cache.Options{
+							SelectorsByObject: cache.SelectorsByObject{
+								&kcorev1.Pod{}: {
+									client.MatchingLabels(tc.labelSelectors),
+									client.MatchingFields(tc.fieldSelectors),
+								},
+							},
+						},
+					)
+					informer, err := builder(cfg, cache.Options{})
+					Expect(err).NotTo(HaveOccurred())
+
+					By("running the cache and waiting for it to sync")
+					go func() {
+						defer GinkgoRecover()
+						Expect(informer.Start(informerCacheCtx)).To(Succeed())
+					}()
+					Expect(informer.WaitForCacheSync(informerCacheCtx)).NotTo(BeFalse())
+
+					gvk := schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Pod"}
+					sii, err := informer.GetInformerForKind(context.TODO(), gvk)
```

Comment:
Nit: This would be easier to read as:
```suggestion
					sii, err := informer.GetInformer(context.TODO(), &corev1.Pod{})
```

Reply:
Done